monitor/statistics.cpp

```diff
@@ -242,9 +242,10 @@ std::string statistics::report(uint64_t categories)
 
 	if (categories & DNET_MONITOR_STATS) {
 #if defined(HAVE_HANDYSTATS) && !defined(HANDYSTATS_DISABLE)
-		rapidjson::Value stats_value(rapidjson::kObjectType);
-		handystats::json::fill(stats_value, allocator, *HANDY_METRICS_DUMP());
-		report.AddMember("stats", stats_value, allocator);
+		rapidjson::Document stats(&allocator);
+		const auto &stats_json = HANDY_JSON_DUMP();
+		stats.Parse<0>(stats_json.c_str());
+		report.AddMember("stats", static_cast<rapidjson::Value &>(stats), allocator);
```

Comment:
Why this cast is required? `rapidjson::Document` is inherited from `rapidjson::Value`, so it looks like the cast should be performed implicitly.

Reply:
May be it is because of old rapidjson version we use, but you variant leads to:
```
../library/libelliptics.so.2.27.4.1: undefined reference to `rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> >::GenericValue(rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> > const&)'
../library/libelliptics.so.2.27.4.1: undefined reference to `rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> >::GenericValue(rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator> > const&)'
```

Reply:
Yep, looks like this template overload is used:
https://github.com/yandex/handystats/blob/1a93b374503cb12c43970ff5d76699becd55c4e0/include/handystats/rapidjson/document.h#L317